src/main/java/org/apache/xml/security/keys/KeyInfo.java

```diff
@@ -361,6 +362,23 @@ public void add(DEREncodedKeyValue derEncodedKeyValue) {
         addReturnToSelf();
     }
 
+    /**
+     * Method add AgreementMethod to the KeyInfo
+     *
+     * @param agreementMethod the AgreementMethod to be added. The AgreementMethod must extend
+     *                        class {@link org.apache.xml.security.utils.ElementProxy}
+     */
+    public void add(AgreementMethod agreementMethod)  {
```

Comment:
Similar comment as my other one. I really think we should try to avoid mixing xml encryption dependencies into this API.

Reply:
I moved the newly added "AgreementMethod methods" to "KeyInfoEncExtension".   Please note that KeyInfo still has "encryption dependencies" due to EncryptionKey. To fully decouple KeyInfo from  encryption package we should move all methods using EncryptionKey to  the KeyInfoEncExtension. Should I move methods for that class as well or this is to big change and might break some implementations?

Reply:
Ah, right - I missed that. I think the `EncryptionKey` methods have been in `KeyInfo` a long time, so removing them with no warning may not be a good idea. I suggest adding similar methods to `KeyInfoEncExtension` and deprecating the ones on `KeyInfo` with a future plan to eventually remove them.